Activities/Module13 Activity.md

```diff
@@ -146,3 +146,217 @@ In case your server is is not responding to requests after a long period of inac
+
+               echo "Server started successfully. Now stopping..."
+               killall node
+
+               npm run test
+
+            - name: Lint backend
+            if: ${{ always() }}
+            run: cd server; npm run lint
+      ```
+
+## Step 4: Connect to GitHub for Continuous Deployment
+
+1. **Connect to GitHub Repository**
+   * In your Heroku app dashboard, scroll to the **Deployment method** section and select **GitHub**.
+   * Click **Connect to GitHub** and search for your repository.
+   * Select the repository containing your backend service code.
+
+   ![Connect to your Repo](Module13Assets/ConnectToGithub.png)
+
+2. **Configure Required Files**
+   * You need to make these changes to the package.json
+     * Move "dotenv" from the "devDependecies" to "dependencies".
+     * Change the "start" script in your server/package.json from "ts-node server.ts" to "node dist/server.js".
+     * Add a new script called "start:dev" with the value "ts-node server.ts". You can use this for your local development.
+   * The package.json file should look something like this:
+     * [`package.json`](Module13Assets/package.json) - Contains required dependencies and scripts
+
+   * You need to make these changes to the tsconfig.json
+     * Go to server/tsconfig.json and scroll to the end. Modify the "include" and "exclude" values, as specified in the tsconfig.json:
+     ```json
+      "include": ["./**/*.ts"],
+      "exclude": ["node_modules", "dist"]
+     ```
+   * The tsconfig.json file should look something like this:
+     * [`tsconfig.json`](Module13Assets/tsconfig.json) - TypeScript configuration
+
+## Step 4: Deploy the Code
+
+1. **Manual Deployment**
+   * Configure the Manual deploy option in Heroku. Everytime you push a change to your main branch, you will have to login to Heroku and click on the Deploy Branch button.
+   * In Heroku, scroll to the **Manual deploy** section.
+   * Choose the main (or preferred deployment) branch.
+   * Click **Deploy Branch** to deploy your application manually.
+
+    ![Manual Deploy](Module13Assets/ManualDeployment.png)
+
+2. **Automatic Deployment Option**
+   * Configure the Automatic deploy option in Heroku. Everytime you push a change to your main branch, Heroku will automatically publish your changes.
+   * In Heroku, scroll to the **Automatic deploys** section.
+   * Choose the main (or preferred deployment) branch.
+   * Click **Enable Automatic Deploys** to deploy your application automatically.
+
+## Step 5: Environment Variables
+
+1. **Set Up Environment Variables**
+   * In your Heroku dashboard, go to **Settings -> Config Vars**
+   * Add your MongoDB connection string as a config var:
+     ```
+     Key: MONGODB_URI
+     Value: <your MongoDB connection string>
+
+     Key: PORT
+     Value: 8000
+     ```
+   * Make sure your MongoDB connection matches the URI in your Mongo Cloud.
+   * Another option is to add the .env file to your server repository.
```

Comment:
This is not possible as the `.gitignore` will ignore `.env` files. Remove this step/point.

Reply:
Done